File: src/main/java/handler/ProjectionsHandler.java
package handler;

import api.ApiCaller;
import api.DataCollector;
import collect.stats.MLBProjections;
import collect.stats.NFLProjections;
import collect.stats.NHLProjections;
import collect.stats.NBAProjections;
import util.AWSClient;

import java.util.HashMap;
import java.util.Map;

public class ProjectionsHandler {
    private DataCollector dataCollector = new DataCollector(new ApiCaller());
    private NBAProjections nbaProjections = new NBAProjections(dataCollector);
    private NFLProjections nflProjections = new NFLProjections(dataCollector);
    private NHLProjections nhlProjections = new NHLProjections(dataCollector);
    private MLBProjections mlbProjections = new MLBProjections(dataCollector);
    private AWSClient AWSClient = new AWSClient();

    public Map<String, Object> handleRequest(Map<String, String> input) {
        String sport = input.get("sport");
        String invocationType = input.getOrDefault("invocationType", "web");
        Map<Integer, Map<String, Object>> result =
                (sport.equals("mlb")) ? mlbProjections.getProjectionsData() :
                        sport.equals("nfl") ? nflProjections.getProjectionsData() :
                                (sport.equals("nhl")) ? nhlProjections.getProjectionsData() :
                                        nbaProjections.getProjectionsData();
        Map<String, Object> resultMap = new HashMap<>();
        if (invocationType.equals("pipeline")) {
            AWSClient.uploadToS3(sport + "ProjectionsData.json", result);
            resultMap.put("sport", sport);
        }
        else {
            resultMap.put("body", result);
        }
        return resultMap;
    }
}

File: src/main/kotlin/handler/ProjectionsHandler.kt
package handler

import collect.stats.MlbProjections
import collect.stats.NbaProjections
import collect.stats.NflProjections
import collect.stats.NhlProjections
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import util.AWS

class ProjectionsHandler {
    suspend fun handleRequest(input: Map<String, String>): Map<String, Any> {
        val sport = input.getValue("sport")
        val invocationType = input.getOrDefault("invocationType", "web")
        val projections: Map<Int, Map<String, Any?>> =
                when (sport) {
                    "mlb" -> getMlbProjectionsData()
                    "nfl" -> getNflProjectionsData()
                    "nhl" -> getNhlProjectionsData()
                    "nba" -> getNbaProjectionsData()
                    else -> mapOf()
                }
        return if (invocationType == "pipeline") {
            uploadToS3("${sport}ProjectionsData.json", projections)
            mapOf("sport" to sport)
        } else mapOf("body" to projections)
    }

    suspend fun getMlbProjectionsData(): Map<Int, Map<String, Any?>> {
        return MlbProjections().getMlbProjectionsData()
    }

    suspend fun getNflProjectionsData(): Map<Int, Map<String, Any?>> {
        return NflProjections().getNflProjectionsData()
    }

    suspend fun getNhlProjectionsData(): Map<Int, Map<String, Any?>> {
        return NhlProjections().getNhlProjectionsData()
    }

    suspend fun getNbaProjectionsData(): Map<Int, Map<String, Any?>> {
        return NbaProjections().getNbaProjectionsData()
    }

    suspend fun uploadToS3(fileName: String, objectToUpload: Any) {
        return withContext(Dispatchers.Default) { AWS().uploadToS3(fileName, objectToUpload) }
    }
}
